File: backend/src/main/scala/slaydemo/backend/battle/objects/BattleParticipantKind.scala
package slaydemo.backend.battle.objects

enum BattleParticipantKind {
  case Human
  case Bot
}

object BattleParticipantKind {
  def fromBotFlag(isBot: Boolean): BattleParticipantKind =
    if isBot then BattleParticipantKind.Bot else BattleParticipantKind.Human

  def isBot(value: BattleParticipantKind): Boolean =
    value == BattleParticipantKind.Bot
}

File: frontend/src/objects/battle/microservices/actors/objects/player/BattleParticipantKind.ts
export type BattleParticipantKind = "Human" | "Bot";

export function battleParticipantKindFromBotFlag(isBot: boolean): BattleParticipantKind {
  return isBot ? "Bot" : "Human";
}

export function battleParticipantKindIsBot(value: BattleParticipantKind): boolean {
  return value === "Bot";
}

